### PR TITLE
(DOCSP-11469): Add TypeScript to Realm Node QuickStart

### PR DESCRIPTION
**Pull request by nlarew**
## Description

- Tweaked code style in JS snippets & full example
- Added TypeScript snippets & full example

## Jira

- (DOCSP-11469): Add TypeScript to Realm Node QuickStart

## Staged Changes

- [Node > Quick Start](https://docs-mongodbcom-staging.corp.mongodb.com/realm/nlarew/node-qs-typescript/node/quick-start.html)

### CODE REVIEW

File: source/node/examples/node-quickstart.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  // Add a couple of Tasks in a single, atomic transaction
+  // Realm automatically sets the _partition property based on the partitionValue used to open the realm
+  realm.write(() => {
+    const task1: Task = realm.create("Task", {
+      _id: new ObjectId(),
+      name: "go grocery shopping",
+      status: "Open",
+    });
+
+    const task2: Task = realm.create("Task", {
```
Comment (cbush) [inline source/node/examples/node-quickstart.ts:52]: Explicit type is understandable for a demo, but we already know it's a task from above. Prefer type inference?
```suggestion
    const task2 = realm.create("Task", {
```

File: source/node/examples/node-quickstart.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  console.error(err);
+});
+
+// Define the collection notification listener
+const taskListener: Realm.CollectionChangeCallback<Task> = (
+  tasks: Realm.Collection<Task>,
+  changes: Realm.ObjectChanges
+) => {
+  // Update UI in response to deleted objects
+  (changes.deletions as unknown as number[]).forEach((index) => {
```
Comment (cbush) [inline source/node/examples/node-quickstart.ts:89]: What's with the cast?

Comment (nlarew) [reply source/node/examples/node-quickstart.ts:89]: Workaround for this TS type bug, which will hopefully go away soon! https://jira.mongodb.org/browse/RJS-709

File: source/node/examples/node-quickstart.ts (hunk 1 above)
Comment (nlarew) [reply source/node/examples/node-quickstart.ts:52]: Yeah my thought here was demo -> more explicit = more better, but why not show both explicit and inferred typing?

**Pull request merged**